### DISCUSSION TITLE
`go to defintion` (`gd`) does not work in typescript

### DISCUSSION THREAD

**Discussion by rykugur** [LazyVim/LazyVim#2433] [Q&A] [answered]
While working in a typescript file, pressing `gd` on a type will bring me to its import line. Not the end of the world, but I expected to be brought to its actual definition.

I read a few posts online and seems that this is just the way it is, and pressing `gd` again should bring me to the actual definition (or index file) - fine, I can deal with that. However, it is not working for me. On the second `gd` invocation, nothing happens.

I do have the typescript extras enabled:

![image](https://github.com/LazyVim/LazyVim/assets/2609898/d3b18576-70c6-4744-8a1a-386f48d36e63)

Is there something else I need to enable or some other way to accomplish this? it's one of the few things holding me back from switching over completely from VS Code.

**Comment by rykugur** [comment, marked as answer]
I think I got this working. In my case, `npm` was missing.

Why npm is necessary for that? Dunno.

**Reply by 0xblur** [reply]
TypeScript LSP sends me to `index.d.ts` when `gd`, `gI` or `gy` and I'm looking how to fix this issue. What you meant by `npm` was missing?

**Reply by rykugur** [reply]
> TypeScript LSP sends me to `index.d.ts` when `gd`, `gI` or `gy` and I'm looking how to fix this issue. What you meant by `npm` was missing?

Context: I run nixOS. `npm` was not on my path and it appears the plugin/LSP require(d/s) it.
